app/code/Magento/CatalogUrlRewrite/Plugin/Eav/AttributeSetRepository/RemoveProductUrlRewrite.php

```diff
@@ -0,0 +1,83 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+
+namespace Magento\CatalogUrlRewrite\Plugin\Eav\AttributeSetRepository;
+
+use Magento\Catalog\Model\ResourceModel\Product\Collection;
+use Magento\Catalog\Model\ResourceModel\Product\CollectionFactory;
+use Magento\CatalogUrlRewrite\Model\ProductUrlRewriteGenerator;
+use Magento\Eav\Api\AttributeSetRepositoryInterface;
+use Magento\Eav\Api\Data\AttributeSetInterface;
+use Magento\UrlRewrite\Model\UrlPersistInterface;
+use Magento\UrlRewrite\Service\V1\Data\UrlRewrite;
+
+/**
+ * Remove url rewrites for products with given attribute set.
+ */
+class RemoveProductUrlRewrite
+{
+    /**
+     * @var int
+     */
+    private $chunkSize = 1000;
+
+    /**
+     * @var UrlPersistInterface
+     */
+    private $urlPersist;
+
+    /**
+     * @var CollectionFactory
+     */
+    private $collectionFactory;
+
+    /**
+     * ProductUrlRewriteProcessor constructor.
+     *
+     * @param UrlPersistInterface $urlPersist
+     * @param CollectionFactory $collectionFactory
+     */
+    public function __construct(UrlPersistInterface $urlPersist, CollectionFactory $collectionFactory)
+    {
+        $this->urlPersist = $urlPersist;
+        $this->collectionFactory = $collectionFactory;
+    }
+
+    /**
+     * Remove url rewrites for products with given attribute set.
+     *
+     * @param AttributeSetRepositoryInterface $subject
+     * @param \Closure $proceed
+     * @param AttributeSetInterface $attributeSet
+     * @return bool
+     *
+     * @SuppressWarnings(PHPMD.UnusedFormalParameter)
+     */
+    public function aroundDelete(
```

Comment:
Use after plugin instead of around

Reply:
Sorry, may be I have missed something, but if I'll use after plugin, I won't have attribute set in result, only bool true, and instance of AttrubuteSetRepository. And I need attribute set id in order to filter product collection.

Reply:
Sorry, may be I have missed something. But, if I'll use after plugin I won't have attribute set id in results, only bool true and instance of AttributeSetRepository. And I need attribute set id in order to filter product collection.